source/memory.tex

```diff
@@ -799,6 +817,83 @@
 \end{note}
 \end{itemdescr}
 
+\rSec2[obj.lifetime]{Explicit lifetime management}
+
+\indexlibraryglobal{start_lifetime_as}%
+\begin{itemdecl}
+template<class T>
+  T* start_lifetime_as(void* p) noexcept;
+template<class T>
+  const T* start_lifetime_as(const void* p) noexcept;
+template<class T>
+  volatile T* start_lifetime_as(volatile void* p) noexcept;
+template<class T>
+  const volatile T* start_lifetime_as(const volatile void* p) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+\tcode{T} is an implicit-lifetime type.
+
+\pnum
+\expects
+\tcode{[p, (char*)p + sizeof(T))} denotes a region of allocated storage
+that is
+a subset of the region of storage
+reachable through\iref{basic.compound} \tcode{p} and
+suitably aligned for the type \tcode{T}.
+
+\pnum
+\effects
+Implicitly creates objects\iref{intro.object} within the denoted region
+consisting of an object \placeholder{a} of type \tcode{T}
+whose address is \tcode{p} and
+objects nested within \placeholder{a}
+as follows:
+The object representation of \placeholder{a}
+is the contents of the storage prior to the call to \tcode{start_lifetime_as}.
+The value of each created object \placeholder{o}
+of trivially-copyable type \tcode{U}
+is determined in the same manner as for a call
+to \tcode{bit_cast<U>(E)}\iref{bit.cast},
+where \tcode{E} is an lvalue of type \tcode{U} denoting \placeholder{o},
+except that the storage is not accessed.
+The value of any other created object is unspecified.
+\begin{note}
+The unspecified value can be indeterminate.
+\end{note}
+
+\pnum
+\returns
+% FIXME: We should introduce "a" outside of the \effects clause
+A pointer to the \placeholder{a} defined in the Effects clause above.
+\end{itemdescr}
+
+\indexlibraryglobal{start_lifetime_as_array}%
+\begin{itemdecl}
+template<class T>
+  T* start_lifetime_as_array(void* p, size_t n) noexcept;
+template<class T>
+  const T* start_lifetime_as_array(const void* p, size_t n) noexcept;
+template<class T>
+  volatile T* start_lifetime_as_array(volatile void* p, size_t n) noexcept;
+template<class T>
+  const volatile T* start_lifetime_as_array(const volatile void* p, size_t n) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+\tcode{n > 0} is \tcode{true}.
+
+\pnum
+\effects
+Equivalent to:
+\tcode{return *start_lifetime_as<U>(p);}
+where \tcode{U} is the type ``array of \tcode{n T}''.
```

Comment:
```suggestion
where \tcode{U} is the type ``array of \tcode{n} \tcode{T}''.
```

Reply:
done, but why is this better?

Reply:
I don't know. That's just how all phrases starting with "``array of " are formatted.